plugins/web_adapter/lib/src/html/adapter.dart

```diff
@@ -65,18 +66,35 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
     final completer = Completer<ResponseBody>();
 
     xhr.onLoad.first.then((_) {
-      final Uint8List body = (xhr.response as ByteBuffer).asUint8List();
-      completer.complete(
-        ResponseBody.fromBytes(
-          body,
-          xhr.status!,
-          headers: xhr.responseHeaders.map((k, v) => MapEntry(k, v.split(','))),
-          statusMessage: xhr.statusText,
-          isRedirect: xhr.status == 302 ||
-              xhr.status == 301 ||
-              options.uri.toString() != xhr.responseUrl,
-        ),
-      );
+      if (options.responseType == ResponseType.blobUrl) {
+        completer.complete(
+          ResponseBody.fromString(
+            Url.createObjectUrl(xhr.response),
```

Comment:
Could you reuse other fields by constructing the `ResponseBody` with the unnamed constructor?

Reply:
The `ResponseBody` with the unnamed constructor requires steam, we have a url string !